tasty-query/shared/src/test/scala/tastyquery/SubtypingSuite.scala

```diff
@@ -830,4 +834,228 @@ class SubtypingSuite extends UnrestrictedUnpicklingSuite:
+    val matchTypeMono2SubtypeTypeMember = matchTypeMono2Subtype.appliedTo(thisTypeMemberRef).asInstanceOf[MatchType]
+    assert(clue(matchTypeMono2SubtypeTypeMember.reduced).isEmpty)
+
+    assertStrictSubtype(matchTypeMono2SubtypeTypeMember, matchTypeMono2TypeMember)
+      .withRef[MatchTypeMono2Subtype[TypeMember], MatchTypeMono2[TypeMember]]
+
+    val matchTypePoly2 = findTypesFromTASTyNamed(typeName("MatchTypePoly2"))
+    val matchTypePoly2TypeMember = matchTypePoly2.appliedTo(thisTypeMemberRef).asInstanceOf[MatchType]
+    assert(clue(matchTypePoly2TypeMember.reduced).isEmpty)
+
+    val matchTypePoly2TypeMemberBis = matchTypePoly2.appliedTo(thisTypeMemberRef).asInstanceOf[MatchType]
+
+    assertEquiv(matchTypePoly2TypeMember, matchTypePoly2TypeMemberBis)
+      .withRef[MatchTypePoly2[TypeMember], MatchTypePoly2[TypeMember]]
+    assertNeitherSubtype(matchTypePoly2TypeMember, matchTypeMono2TypeMember)
+      .withRef[MatchTypePoly2[TypeMember], MatchTypeMono2[TypeMember]]
+  }
+
+  testWithContext("match-types-with-literal-types") {
+    val typesFromTasty = new subtyping.TypesFromTASTy
+    import typesFromTasty.*
+
+    val matchTypeLiterals = findTypesFromTASTyNamed(typeName("MatchTypeLiterals"))
+
+    def const(x: Int): ConstantType = ConstantType(Constant(x))
+
+    assertEquiv(matchTypeLiterals.appliedTo(const(1)), defn.IntType).withRef[MatchTypeLiterals[1], Int]
+    assertEquiv(matchTypeLiterals.appliedTo(const(3)), defn.StringType).withRef[MatchTypeLiterals[3], String]
+    assertEquiv(matchTypeLiterals.appliedTo(const(5)), defn.BooleanType).withRef[MatchTypeLiterals[5], Boolean]
+
+    assert(clue(matchTypeLiterals.appliedTo(const(2)).asInstanceOf[MatchType].reduced).isEmpty)
+  }
+
+  testWithContext("match-types-with-enums") {
+    import subtyping.TypesFromTASTy.MyEnum
+
+    val typesFromTasty = new subtyping.TypesFromTASTy
+    import typesFromTasty.*
+
+    val matchTypeEnums = findTypesFromTASTyNamed(typeName("MatchTypeEnums"))
+    val MyEnumPrefix = "subtyping.TypesFromTASTy.MyEnum"
+
+    val ParametricTycon = ctx.findStaticType(s"$MyEnumPrefix.Parametric").staticRef
+
+    assertEquiv(matchTypeEnums.appliedTo(ctx.findStaticTerm(s"$MyEnumPrefix.Singleton1").staticRef), defn.IntType)
+      .withRef[MatchTypeEnums[MyEnum.Singleton1.type], Int]
+
+    assertEquiv(matchTypeEnums.appliedTo(ParametricTycon.appliedTo(defn.IntType)), defn.DoubleType)
+      .withRef[MatchTypeEnums[MyEnum.Parametric[Int]], Double]
+
+    assertEquiv(matchTypeEnums.appliedTo(ctx.findStaticTerm(s"$MyEnumPrefix.Singleton2").staticRef), defn.StringType)
+      .withRef[MatchTypeEnums[MyEnum.Singleton2.type], String]
+
+    assertEquiv(matchTypeEnums.appliedTo(ParametricTycon.appliedTo(defn.StringType)), defn.StringType)
+      .withRef[MatchTypeEnums[MyEnum.Parametric[String]], String]
+  }
+
+  testWithContext("match-types-with-old-style-enums") {
+    import subtyping.TypesFromTASTy.MyEnum
```

Comment:
```suggestion
    import subtyping.TypesFromTASTy.OldStyleEnum
```
plus update the other references

Reply:
Oops, indeed. Fixed.